cmd/cagent/main.go

```diff
@@ -481,17 +481,19 @@ func writePidFileIfNeeded(ca *cagent.Cagent, oneRunOnlyModePtr *bool) {
 	if ca.Config.PidFile != "" && !*oneRunOnlyModePtr && runtime.GOOS != "windows" {
 		err := ioutil.WriteFile(ca.Config.PidFile, []byte(strconv.Itoa(os.Getpid())), 0664)
 		if err != nil {
-			log.Errorf("Failed to write pid file at: %s", ca.Config.PidFile)
+			log.WithFields(log.Fields{
+				"pidFile": ca.Config.PidFile,
+			}).WithError(err).Errorf("Failed to write pid file")
 		}
 	}
 }
 
 func removePidFileIfNeeded(ca *cagent.Cagent, oneRunOnlyModePtr *bool) {
 	if ca.Config.PidFile != "" && !*oneRunOnlyModePtr && runtime.GOOS != "windows" {
 		err := os.Remove(ca.Config.PidFile)
-		if err != nil {
-			log.Errorf("Failed to remove pid file at: %s", ca.Config.PidFile)
-		}
+		log.WithFields(log.Fields{
```

Comment:
Not sure why the condition was removed:
`if err != nil {`

Reply:
That was typo, thank you for vigilance.